lib/core/text/flat.nit

```diff
@@ -1148,8 +1148,9 @@ redef class Array[E]
 	do
 		var l = length
 		if l == 0 then return ""
-		if l == 1 then if self[0] == null then return "" else return self[0].to_s
-		var its = _items
+		var first = self[0]
+		if l == 1 then if first == null then return "" else return first.to_s
+		var its = _items.as(not null)
```

Comment:
Come to think of it, having its declared before the ifs, and replacing the `self[0]` by `its[0]` could be a small optimization :)